### PR TITLE
use authenticated entryServerNode to know who is primary/backup + upd…

### PR DESCRIPTION
**Pull request by yossipapi**
…ate client libs

### CODE REVIEW

File: lib/BackendClient.js [hunk 1]
```diff
@@ -202,10 +232,12 @@ var BackendClient = (function(){
             });
     };
 
-    BackendClient.getEntries = function(entryList) {
+    BackendClient.getEntries = function(entryId) {
         return createSession()
             .then(function () {
-                return getEntries(entryList);
+                return getMediaServerIndexPromise().then(function(index) {
```
Comment (kalturaguy) [inline lib/BackendClient.js]: change from index to Id

File: lib/BackendClient.js [hunk 2]
```diff
@@ -69,6 +74,35 @@ var BackendClient = (function(){
         return deferred.promise;
     }
 
+    function getMediaServerIndexPromise() {
```
Comment (kalturaguy) [inline lib/BackendClient.js]: change to getMediaServerIndexPromise

File: lib/BackendClient.js [hunk 3]
```diff
@@ -124,39 +158,35 @@ var BackendClient = (function(){
...
+
+            if (result.objectType !== "KalturaAPIException") {  //Use to indicate success operation
+                moduleLogger.info("Got server node info for entry %s response: %j (took %d)", entryId, result, operationTime);
+                printResponseHeaders(moduleLogger, headers);
+                var liveEntry = convertKalturaEntry(result);
+
+                if(result.primaryServerNodeId === mediaServerIndex)
+                    liveEntry.serverType = kalturaTypes.KalturaEntryServerNodeType.LIVE_PRIMARY;
+                else
+                    liveEntry.serverType = kalturaTypes.KalturaEntryServerNodeType.LIVE_BACKUP;
```
Comment (kalturaguy) [inline lib/BackendClient.js]: rename serverType to serverNodeType

File: lib/BackendClient.js [hunk 4]
```diff
@@ -178,7 +208,7 @@ var BackendClient = (function(){
         return deferred.promise;
     }
 
-    function unregisterEntryInDatabase(entryId, logger) {
+    function unregisterEntryInDatabase(entryId, serverIndex, logger) {
```
Comment (kalturaguy) [inline lib/BackendClient.js]: rename serverIndex to entryServerNodeType

Comment (yossipapi) [inline lib/BackendClient.js]: Its called entryType on the api object this is why I called it serverType

**Pull request merged**